numpy-stubs/__init__.pyi

```diff
@@ -402,14 +402,15 @@ class number(generic):
 class bool_(_real_generic): ...
 class object_(generic): ...
 
-class datetime64(_real_generic):
+class datetime64:
     @overload
     def __init__(
         self, _data: Union[datetime64, str, dt.datetime] = ..., _format: str = ...
     ) -> None: ...
     @overload
     def __init__(self, _data: int, _format: str) -> None: ...
-    def __sub__(self, other: datetime64) -> timedelta64: ...
+    def __add__(self, other: Union[timedelta64, int]) -> datetime64: ...
+    def __sub__(self, other: Union[timedelta64, datetime64, int]) -> timedelta64: ...
```

Comment:
Do we need an overload here? `datetime64 - datetime64 = timedelta64`, but `datetime64 - timedelta/int = datetime64`, right?

Reply:
You're totally correct - I will add the overloads.